### ISSUE TITLE
Hi author, 66% accuracy of the paper results cannot reproduce the conclusion

### ISSUE THREAD

**Issue by 18001178267** [totruongan/Robust-Automatic-Modulation-Classification-in-Low-Signal-to-Noise-Ratio#2]
Run in the same local environment once, with an average accuracy of 50-60; At the same time, I looked at the ipynb file retained in the Codebase, and the precision is also 50+, which does not reach the conclusion of the paper by 66%. If the author has time, could you please upload a ipynb file which has 66% accuracy? Thanks

**Comment by totruongan** [comment]
Hi, The result will be based on the local environment and the CPU power. 
please give me your mail I will send the training Ipynb for you. 
Thanks.

**Comment by totruongan** [comment]
![image](https://github.com/totruongan/Robust-Automatic-Modulation-Classification-in-Low-Signal-to-Noise-Ratio/assets/95015972/dc91c0ef-69c2-45de-b576-b21cf09d5dba)
Hello, there's already an ipynb file in the code folder.
Please note that the average accuracy for all SNR values of the mode is 66%, so please keep that in mind.l.

**Comment by 18001178267** [comment]
hi, I found the file you mentioned, which randomly divides 220000 samples into training and validation sets in an 8:2 ratio. However, I found that in the training file, before the early stop,the val accuracy is only 50+%, not close to 66%; Simultaneously running mcldnn and pet-cgdnn mentioned in the paper, the val accuracy in the figure are close to 60%. My local configuration is TensorFlow2.9 and GPU 3080.
![image](https://github.com/totruongan/Robust-Automatic-Modulation-Classification-in-Low-Signal-to-Noise-Ratio/assets/54790789/e8d77572-419b-4b2b-bf0d-2792304be424)
![image](https://github.com/totruongan/Robust-Automatic-Modulation-Classification-in-Low-Signal-to-Noise-Ratio/assets/54790789/1e4e4e9d-b82a-443a-889e-93dba9045e45)

**Comment by totruongan** [comment]
66% is the average accuracy of the model. it is not the val accuracy. you should calculate **the average of the accuracy of models for all SNR value.** 
Totruonganspk@gmail.com. if you have any code problem send me an email and your code in google colab.

**Comment by 18001178267** [comment]
bro, your training set is used for both training and evaluation, it doesn't make sense
![image](https://github.com/totruongan/Robust-Automatic-Modulation-Classification-in-Low-Signal-to-Noise-Ratio/assets/54790789/7f45ef6a-5dbd-452c-8e67-e9805c582cd1)
You should only use validation sets to evaluate the model instead of full data。

**Comment by 18001178267** [comment]
Other models and evaluation methods are reasonable, only using validation sets for evaluation
![image](https://github.com/totruongan/Robust-Automatic-Modulation-Classification-in-Low-Signal-to-Noise-Ratio/assets/54790789/0e64f41c-23a0-4b14-86dd-6a16adfec43d)
![image](https://github.com/totruongan/Robust-Automatic-Modulation-Classification-in-Low-Signal-to-Noise-Ratio/assets/54790789/21072b14-9694-41ee-a728-778cbcc4aea1)

**Comment by totruongan** [comment]
pls re-read the code. 
Thanks

**Comment by totruongan** [comment]
Hi, I would like to make it clear to you. I use training data for training and I use test datasets for testing and evaluating.
